src/main/java/demon/Commands.java

```diff
@@ -0,0 +1,285 @@
+package demon;
+
+import java.io.IOException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * This class serves as a central point for managing and executing a variety of user commands.
+ * It encapsulates the logic for each command as individual methods, allowing for easy execution
+ * based on user input. Commands such as 'list', 'unmark', 'mark', 'delete', and others are supported, each
+ * corresponding to a specific method within this class.
+ */
+public class Commands {
+    Storage storage;
+
+    public Commands(String filePath) {
+        this.storage = new Storage(filePath);
+    }
+    /**
+     * Prints out all tasks saved numbered in sequence.
+     * No tasks will be printed if there is no saved tasks.
+     *
+     * @param tasks an array containing all tasks.
+     */
+    public void list(TaskList tasks) {
+        Ui.printDivider();
+        System.out.print("List of things to do :\n");
+        for (int i = 1; i <= tasks.getTaskList().size(); i++) {
+            Task item = tasks.getTaskList().get(i - 1);
+            System.out.println("\t" + i + "." + item.toString());
+        }
+        Ui.printDivider();
+    }
+
+    /**
+     * Marks the task in the list as 'not done' if the task was previously marked as done.
+     * If the task was marked as 'not done', then executing this method will not change anything.
+     *
+     * @param tasks an array containing all tasks.
+     * @param num array index of the task list
+     */
+    public void unmark(TaskList tasks, int num) {
+        if (num < 1 | num > tasks.getTaskList().size()) {
+            System.out.println(Ui.notWithinRange() + tasks.getTaskList().size());
+        } else {
+            Task item = tasks.getTaskList().get(num-1);
+            if (item.getStatusIcon().equals("X")) {
```

Comment:
"X" seems to be a magic value that relies on the implementation of Task. It might be easier (and safer) to abstract this out to another boolean method such as `item.isMarked()` that returns true if the item is indeed marked, and false otherwise?